apis/index.js

```diff
@@ -86,7 +90,46 @@ export function fetchLocation(locationKey) {
     .then((location) => location.val());
 }
 
-export async function createLocation(regionKey, options, key) {
+export async function updateLocation(regionKey, key, newLocation) {
+  initialize();
+
+  let pushed;
+  if (key) {
+    pushed = getRef(`locations/${key}`);
+  } else {
+    pushed = pushRef('locations');
+  }
+
+  try {
+    const existing = await fetchLocation(key);
```

Comment:
Can we keep the alert out of the api call? Feels like we are mixing up the layers in odd ways.

Reply:
I moved it out to the `pushLocation` function in location actions. Not sure really where the best place for it to live is honestly.